airbyte-webapp/src/components/ui/Flex/FlexContainer.tsx

```diff
@@ -0,0 +1,72 @@
+import classNames from "classnames";
+import React, { HTMLAttributes } from "react";
+
+import styles from "./FlexContainer.module.scss";
+
+interface FlexContainerProps {
+  className?: string;
+  /**
+   * The flex-direction css property
+   */
+  direction?: "row" | "column" | "row-reverse" | "column-reverse";
+  /**
+   * gap between the flex items - defaults to `md` if not provided. None means no gap is applied, the others map to the respective scss spacing variable.
+   */
+  gap?: "none" | "xs" | "sm" | "md" | "lg" | "xl" | "2xl";
+  /**
+   * The align-items css property
+   */
+  alignItems?: "flex-start" | "flex-end" | "center" | "baseline" | "stretch";
+  /**
+   * The justify-content css property
+   */
+  justifyContent?: "flex-start" | "flex-end" | "center" | "space-between" | "space-around" | "space-evenly";
+}
+
+/**
+ * Renders a div element which layouts its children as flex items as specified by the props.
+ *  Children of a `FlexContainer` can but don't have to be `FlexItem` elements.
+ */
+export const FlexContainer: React.FC<React.PropsWithChildren<FlexContainerProps & HTMLAttributes<HTMLDivElement>>> = ({
+  className,
+  direction = "row",
+  gap = "md",
+  alignItems = "stretch",
+  justifyContent = "flex-start",
+  children,
+  ...otherProps
+}) => {
+  const fullClassName = classNames(
```

Comment:
Is there a benefit to memoizing the className generation?

Reply:
As it depends on so many props my feeling was that the dirty-check would take longer than re-evaluating the className.